system/security/LdapSecurity/ldapconnection.cpp

```diff
@@ -4185,13 +4198,17 @@ class CLdapClient : implements ILdapClient, public CInterface
 
         if(m_ldapconfig->getServerType() != ACTIVE_DIRECTORY)
         {
-            StringBuffer act_dn;
-            if(action.m_account_type == GROUP_ACT)
-                getGroupDN(action.m_account_name.str(), act_dn);
-            else
-                getUserDN(action.m_account_name.str(), act_dn);
-            
-            action.m_account_name.clear().append(act_dn.str());
+            //if not already fully qualified
+            if (0 == strstr(action.m_account_name, "dc=") &&
+                0 == strstr(action.m_account_name, "ou="))
+            {
+                StringBuffer act_dn;
+                if(action.m_account_type == GROUP_ACT)
+                    getGroupDN(action.m_account_name.str(), act_dn);
+                else
+                    getUserDN(action.m_account_name.str(), act_dn);
+                action.m_account_name.clear().append(act_dn.str());
```

Comment:
Just a note that `s.set(str)` is equivalent to `s.clear().append(str);` so you could do:

`action.m_account_name.set(act_dn);`